packages/broker/src/ConfigWizard.ts

```diff
@@ -82,200 +84,164 @@ const DefaultConfig: Config = {
+        when: (answers: inquirer.Answers) => {
+            return answers.generateOrImportEthereumPrivateKey === 'Import'
+        },
+        validate: (input: string): string | boolean => {
             try {
-                wallet = new Wallet(privateKey)
-                this.config.ethereumPrivateKey = wallet.privateKey
-            } catch (privateKeyError) {
-                throw new Error(`Invalid privateKey provided for import: ${privateKey}`)
+                new Wallet(input)
+                return true
+            } catch (e) {
+                if (e.message.includes('invalid hexlify value')){
+                    return `Invalid privateKey provided for import: ${input}`
+                } else {
+                    return e.message
+                }
             }
-
-        } else {
-            throw new Error(`Invalid option ${generateOrImport} provided`)
         }
-
-        return wallet
     }
+]
 
-    async selectPlugins(): Promise<Array<string>>{
-        const plugins: inquirer.Answers = await this.inquirerSinglePrompt( {
-            type: 'checkbox',
-            name:'selectedItems',
-            message: 'Select the plugins to enable',
-            choices: [
-                {name: 'Websocket'},
-                {name:'MQTT'},
-                {name:'HttpPublish'}
-            ]
-        })
+const pluginTemplates = {
+    websocket: { port: DEFAULT_WS_PORT },
+    mqtt: { port: DEFAULT_MQTT_PORT },
+    publishHttp: { port: DEFAULT_HTTP_PORT }
+}
 
-        const selectedPlugins = []
-        for (let i = 0; i < plugins.length; i++) {
-            selectedPlugins.push(plugins[i])
-            if (plugins[i] === 'Websocket') {
-                const wsPort = await this.promptNumberWithDefault({
-                    type: 'input',
-                    name: 'wsPort',
-                    message: `Select a port for the Websocket Plugin [Enter for default: ${DEFAULT_WS_PORT}]`,
-                }, DEFAULT_WS_PORT)
-                this.config.plugins['websocket'] = {
-                    port: wsPort,
-                    payloadMetadata: false,
-                    sslCertificate: null
-                }
-            }
+const pluginSelectorPrompt = {
+    type: 'checkbox',
+    name:'selectPlugins',
+    message: 'Select the plugins to enable',
+    choices: Object.keys(pluginTemplates)
+}
 
-            if (plugins[i] === 'MQTT') {
-                const mqttPort = await this.promptNumberWithDefault({
-                    type: 'input',
-                    name: 'mqttPort',
-                    message: `Select a port for the MQTT Plugin [Enter for default: ${DEFAULT_MQTT_PORT}]`,
-                }, DEFAULT_MQTT_PORT)
-                this.config.plugins['mqtt'] = {
-                    port: mqttPort,
-                    payloadMetadata: false
-                }
+const pluginPrompts = Object.values(pluginTemplates).map((plugin, i) => {
+    const pluginName = Object.keys(pluginTemplates)[i]
```

Comment:
We can't know the iteration order of the object keys

Reply:
Can't we?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/values

the `for...in` loop for `Object.values` should do the same string sorting as the one provided in `Object.keys`, right?

Reply:
You are right 👍 Wasn't aware of that the matching ordering is a documented feature. 

(Still it would be maybe more readable and more straightforward if it would be something like)
```
Object.keys(pluginTemplates).map((pluginName) => {
  const plugin = pluginTemplates[name]
})
```